src/building/java/net/remmintan/mods/minefortress/blocks/building/FortressBuildingBlockEntity.kt

```diff
@@ -129,6 +129,12 @@ class FortressBuildingBlockEntity(pos: BlockPos?, state: BlockState?) :
     override fun getFurnace(): FurnaceBlockEntity? =
         furnaceBlockPos?.let { this.getWorld()?.let { w -> w.getBlockEntity(it) as? FurnaceBlockEntity } }
 
+    override fun getFurnacePos(): BlockPos? =
+        this.blockData?.referenceState
```

Comment:
This method will be called quite often in the game causing the full reference state loop over and over again. Could you pelase consider caching this value? at least for some minutes or for some hundrets of ticks? or even once per world load... not sure what is the best here.

Reply:
Good point.. I added a `findFurnaces` method which will do this looping. The `findFurnaces` is called on init, and ONLY IF the furnace list is found to be null during the BlacksmithDailyTask. Since `init()` isn't called when loading the world, the `findFurnaces` method has to be called somehow. 

We can keep the list up to date by simply calling it on init() after that though. (From what I can tell). 

Otherwise `getFurnacePos` is called, which has been updated to only return the cached list.